### PR TITLE
feat: introduce cli errors

### PR DESCRIPTION
**Pull request by kimskimchi**
## Summary

Introduce cli errors. These will be used for outputting user facing messages in a consistent format, where as the original errors (`OriginalError`) should be used for debugging. 

See a draft of how this is to be used: #1760

## Checklist

- [x] Wrote appropriate unit tests
- [x] Considered security implications of the change
- [x] Updated associated docs where necessary
- [x] Updated associated configuration where necessary
- [x] Change is backwards compatible if it needs to be (user can upgrade without manual steps?)
- [x] Nothing sensitive logged
- [x] Considered data migrations for smooth upgrades

## Related Issues

Related #1394

### CODE REVIEW

File: internal/cmd/error.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package cmd
+
+import "fmt"
+
+// CLI Errors are user facing errors that is formatted.
```
Comment (BruceMacD) [inline internal/cmd/error.go:5]: nit:
```suggestion
// CLI Errors are user facing errors that are formatted.
```

File: internal/cmd/error.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+// Should be used for communication, rather than a stacktrace.
+type Error struct {
+	// Error that is caused by the system, used for debugging
+	// Only set this if you need it to be printed as part of the messsage
```
Comment (BruceMacD) [inline internal/cmd/error.go:9]:
```suggestion
	// Only set this if you need it to be printed as part of the message
```

File: internal/cmd/error.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	if e.OriginalError != nil {
+		if len(e.Message) == 0 {
+			return fmt.Sprintf("Internal error:\n%v", e.OriginalError)
+		}
+
+		// Strip '.' at the end when message includes the original error
+		if string(e.Message[len(e.Message)-1]) == "." {
+			e.Message = e.Message[:len(e.Message)-1]
+		}
+		return fmt.Sprintf("%v:\n       %v", e.Message, e.OriginalError)
```
Comment (BruceMacD) [inline internal/cmd/error.go:28]: Should this indent be a `\t`?

**Review by BruceMacD — APPROVED** [review]

**Review by dnephin — COMMENTED** [review]
This looks good! I think the only thing that is missing is test cases for `Error` method. We should probably have 3 test cases to show each of the possible formats.

File: internal/cmd/error.go (hunk 3 above)
Comment (kimskimchi) [reply internal/cmd/error.go:28]: \t seems to be off by one space: 
image

**Comment by kimskimchi** [comment]
Will push pr on Monday after we get a couple of test cases in.

**Pull request merged**